docs/programming/profiling-nvidia-gpu-performance.md

```diff
@@ -0,0 +1,1021 @@
+
+The following tools are available for interacting with performance counters:
+
+- `nvidia-smi`: For basic monitoring of GPU resource usage.
+
+- `Nsight Compute (ncu)`: For detailed GPU kernel performance analysis.
+
+- `Nsight Systems (nsys)`: For system-wide GPU and CPU performance tracing.
+
+## Sample CUDA Code: Vector Addition
+This code example will be used throughout this guide to demonstrate how to use each NVIDIA profiling and monitoring tool.
+
+Here’s a simple CUDA program `vectorAdd.cu` that adds two vectors of floats.
+
+```
+#include <iostream>
+#include <cuda_runtime.h>
+
+#define N 1024  // Size of the vectors
+
+__global__ void vectorAdd(float *A, float *B, float *C) {
+    int idx = threadIdx.x + blockIdx.x * blockDim.x;
+    if (idx < N) {
+        C[idx] = A[idx] + B[idx];
+    }
+}
+
+int main() {
+    float *A = new float[N], *B = new float[N], *C = new float[N];
+    for (int i = 0; i < N; i++) {
+        A[i] = static_cast<float>(rand()) / RAND_MAX;
+        B[i] = static_cast<float>(rand()) / RAND_MAX;
+    }
+
+    float *d_A, *d_B, *d_C;
+    size_t size = N * sizeof(float);
+    cudaMalloc(&d_A, size); cudaMalloc(&d_B, size); cudaMalloc(&d_C, size);
+    cudaMemcpy(d_A, A, size, cudaMemcpyHostToDevice);
+    cudaMemcpy(d_B, B, size, cudaMemcpyHostToDevice);
+
+    dim3 block(256);
+    dim3 grid((N + block.x - 1) / block.x);
+    vectorAdd<<<grid, block>>>(d_A, d_B, d_C);
+    cudaMemcpy(C, d_C, size, cudaMemcpyDeviceToHost);
+
+    delete[] A; delete[] B; delete[] C;
+    cudaFree(d_A); cudaFree(d_B); cudaFree(d_C);
+    return 0;
+}
+```
+
+## nvidia-smi
+
+`nvidia-smi` (System Management Interface) is a command-line utility that provides real-time information about GPU utilization, memory usage, temperature, and running processes.
+
+### Getting Started
+
+```
+$ nvidia-smi
+```
+Example output of `nvidia-smi` on the `aa100` partition
+```
++-----------------------------------------------------------------------------------------+
+| NVIDIA-SMI 570.124.06             Driver Version: 570.124.06     CUDA Version: 12.8     |
+|-----------------------------------------+------------------------+----------------------+
+| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
+| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
+|                                         |                        |               MIG M. |
+|=========================================+========================+======================|
+|   0  NVIDIA A100-PCIE-40GB          Off |   00000000:21:00.0 Off |                    0 |
+| N/A   33C    P0             40W /  250W |       1MiB /  40960MiB |      2%      Default |
```

Comment:
These values do not make sense to me (primarily the memory one). I think you only added the process to the table, but did not update these values. Please update these values so they make sense. Additionally, update the table in the `Sensor and Resource Usage Metrics` subsection.